Snakefile

```diff
@@ -139,26 +161,25 @@ rule crispr:
         opentargets_validator --schema {params.schema} {output.evidenceFile}
         """
 
-## epmc                     : processes target/disease evidence strings from ePMC cooccurrence files
-rule epmc:
+rule epmc:                    # Process target/disease evidence strings from ePMC cooccurrence files.
     input:
         inputCooccurences = directory(GS.remote(config['EPMC']['inputBucket']))
     params:
         schema = f"{config['global']['schema']}/opentargets.json"
     output:
-        evidenceFile = GS.remote(f"{config['EPMC']['outputBucket']}/epmc-{timeStamp}.json.gz")
+        evidenceFile = 'epmc.json.gz'
     log:
-        GS.remote(logFile)
+        'log/epmc.log'
     shell:
         """
+        exec &> {log}
         python modules/EPMC.py \
           --cooccurrenceFile {input.inputCooccurences} \
           --output {output.evidenceFile}
         opentargets_validator --schema {params.schema} {output.evidenceFile}
         """
```

Comment:
So the pipeline is still failing. The problem is still the fact that the input is a directory which is not supported for inputs. As a quick workaround, I would suggest that we specify the input as a parameter. 

```python
rule epmc:                    # Process target/disease evidence strings from ePMC cooccurrence files.
    params:
        inputCooccurences = directory(GS.remote(config['EPMC']['inputBucket'])),
        schema = f"{config['global']['schema']}/opentargets.json"
    output:
        evidenceFile = 'epmc.json.gz'
    log:
        'log/epmc.log'
    shell:
        """
        exec &> {log}
        python modules/EPMC.py \
          --cooccurrenceFile {input.inputCooccurences} \
          --output {output.evidenceFile}
        opentargets_validator --schema {params.schema} {params.evidenceFile}
        """
```

As parameters doesn't go through such a checks.

Reply:
This turned out to be a bug in Snakemake, resolved with updating to v7.3.0. In practice, even though the `directory` flag doesn't affect the input section (and in that sense, is "not allowed" there), the actual directory download is still being processed correctly